modules/command-engine/core/src/main/scala/surge/internal/persistence/PersistentActor.scala

```diff
@@ -234,69 +223,85 @@ class PersistentActor[S, M, R, E](
     context.become(freeToProcess(internalActorState))
   }
 
-  private def handle(state: InternalActorState, msg: ProcessMessage[M]): Unit = {
+  private def handle(state: InternalActorState, msg: ProcessMessage[Message]): Unit = {
     context.setReceiveTimeout(Duration.Inf)
     context.become(persistingEvents(state))
 
     processMessage(state, msg)
-      .flatMap {
-        case Left(r) => Future.successful(ACKRejection(r))
-        case Right(handled) =>
+      .flatMap { result =>
+        if (result.isRejected) {
+          // FIXME Temporary no-op publish to trigger side effects later
```

Comment:
This is a bit awkward and warrants some cleanup

Reply:
I'd like to revisit the `PersistentActor` as a whole eventually anyways so it's likely any optimization here now would end up deleted as part of that work. Additionally, we're not currently exposing a model where commands can be explicitly rejected, the current pattern (though maybe a little wacky) is to throw an exception from the command handler and treat it as an error.